fdbserver/worker.actor.cpp

```diff
@@ -185,7 +185,8 @@ ACTOR Future<Void> workerHandleErrors(FutureStream<ErrorInfo> errors) {
 
 			endRole(err.role, err.id, "Error", ok, err.error);
 
-			if (err.error.code() == error_code_please_reboot || err.error.code() == error_code_io_timeout) throw err.error;
+
+			if (err.error.code() == error_code_please_reboot || err.error.code() == error_code_io_timeout || (err.role == Role::SHARED_TRANSACTION_LOG && err.error.code() == error_code_io_error )) throw err.error;
```

Comment:
Should we do the same for SS role as well? 
Although the problem for SS won't cause master recoveries, it will still degrade the performance until the cluster exclude it.

Reply:
There is some thought that the reason the existing behavior was chosen in the first place was to prevent us from immediately trying to reuse a storage disk after it gets an IO error. The current code doesn't completely avoid that possibility, since the process could restart for a variety of reasons, but in many cases an IO error will leave the process running but not operating as a storage server. That generally will give you some time to do replace the machine or take some other action in response to a disk failure.

It may be worth rethinking how we want to do all of this in general, but given this is a somewhat quick patch, it's probably not worth trying to change more behavior than is necessary without thinking through the implications.